src/plugins/management_overview/public/plugin.ts

```diff
@@ -0,0 +1,98 @@
+ */
+
+import { i18n } from '@osd/i18n';
+import {
+  AppMountParameters,
+  CoreSetup,
+  Plugin,
+  DEFAULT_APP_CATEGORIES,
+  CoreStart,
+} from '../../../core/public';
+import { FeatureCatalogueCategory, HomePublicPluginSetup } from '../../home/public';
+import { MANAGEMENT_OVERVIEW_PLUGIN_ID } from '../common/constants';
+import { OverviewApp } from './overview_app';
+
+interface ManagementOverviewSetupDeps {
+  home?: HomePublicPluginSetup;
+}
+
+export interface ManagementOverViewPluginSetup {
+  register: (overviewApp: OverviewApp) => void;
+}
+
+export type ManagementOverViewPluginStart = ManagementOverViewPluginSetup;
+
+/** @public */
+export class ManagementOverViewPlugin
+  implements Plugin<ManagementOverViewPluginSetup, ManagementOverViewPluginStart> {
+  private readonly overviewApps = new Map<string, OverviewApp>();
+
+  private getSortedOverviewApps(): OverviewApp[] {
+    return [...this.overviewApps.values()].sort((a, b) => a.order - b.order);
+  }
+
+  private getRegister() {
+    return (app: OverviewApp) => {
+      if (this.overviewApps.has(app.id)) {
+        throw new Error(
+          `Management overview App tool with id [${app.id}] has already been registered. Use a unique id.`
+        );
+      }
+      this.overviewApps.set(app.id, app);
+    };
+  }
+
+  public setup(
+    coreSetup: CoreSetup,
+    { home }: ManagementOverviewSetupDeps
+  ): ManagementOverViewPluginSetup {
+    const { application, getStartServices } = coreSetup;
+
+    if (home) {
+      home.featureCatalogue.register({
+        id: MANAGEMENT_OVERVIEW_PLUGIN_ID,
+        title: i18n.translate('management.stackManagement.managementLabel', {
+          defaultMessage: 'Management',
+        }),
+        description: i18n.translate('management.stackManagement.managementDescription', {
+          defaultMessage: 'Your center location for managing the OpenSearch Stack.',
+        }),
+        icon: 'managementApp',
+        path: `/app/${MANAGEMENT_OVERVIEW_PLUGIN_ID}`,
+        showOnHomePage: false,
+        category: FeatureCatalogueCategory.ADMIN,
+      });
+    }
+
+    application.register({
+      id: MANAGEMENT_OVERVIEW_PLUGIN_ID,
+      title: i18n.translate('management.overview.overviewTitle', {
+        defaultMessage: 'Overview',
+      }),
+      icon: '/plugins/home/public/assets/logos/opensearch_mark_default.svg',
+      order: 9000,
+      category: DEFAULT_APP_CATEGORIES.management,
+      mount: async (params: AppMountParameters) => {
+        const { element } = params;
+        const [core] = await getStartServices();
+        const overviewApps = this.getSortedOverviewApps();
+
+        const { renderApp } = await import('./application');
+        return renderApp(core, overviewApps, element);
+      },
+    });
+
+    return {
+      register: this.getRegister(),
+    };
+  }
+
+  public start(core: CoreStart): ManagementOverViewPluginStart {
+    return {
+      register: this.getRegister(),
```

Comment:
The reason is that for `dashboards management` and `dev_tools` they may have no sub application installed. For example, in case console is not installed, dev tools will update to hidden and not show up in left navigation.  the logic is placed in start phase, so to avoid register to management overview page at setup phase and then remove it if it's hidden, then i put the register the into start phase also.

Reply:
Let's me follow the pattern and limit the registration only into setup phase